Calculator/calculator.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "eact";
+import Button from "./Button";
+import Display from "./Display";
+import HistoryTable from "./HistoryTable";
+
+const Calculator = () => {
+  const [mode, setMode] = useState("simple");
+  const [num1, setNum1] = useState("");
+  const [num2, setNum2] = useState("");
+  const [operator, setOperator] = useState("");
+  const [result, setResult] = useState("");
+  const [history, setHistory] = useState([]);
+
+  const handleModeChange = () => {
+    setMode(mode === "simple" ? "intermediate" : "simple");
+  };
+
+  const handleNumberClick = (num) => {
+    if (operator) {
+      setNum2(num2 + num);
+    } else {
+      setNum1(num1 + num);
+    }
+  };
+
+  const handleOperatorClick = (op) => {
+    setOperator(op);
+  };
+
+  const handleEqualsClick = () => {
+    let result;
+    switch (operator) {
+      case "+":
+        result = parseFloat(num1) + parseFloat(num2);
+        break;
+      case "-":
+        result = parseFloat(num1) - parseFloat(num2);
+        break;
+      case "*":
+        result = parseFloat(num1) * parseFloat(num2);
+        break;
+      case "/":
+        result = parseFloat(num1) / parseFloat(num2);
+        break;
+      default:
+        result = "";
+    }
+    setResult(result);
+    setHistory([...history, `${num1} ${operator} ${num2} = ${result}`]);
+    setNum1("");
+    setNum2("");
+    setOperator("");
+  };
+
+  const handleClearClick = () => {
+    setNum1("");
+    setNum2("");
+    setOperator("");
+    setResult("");
+  };
+
+  const getButtons = () => {
+    if (mode === "simple") {
+      return [
+        { label: "1", onClick: () => handleNumberClick("1") },
```

Comment:
Please Create a generic Function , maybe use an array for your numbers , this approach seems redundant